### ISSUE TITLE
Inconsistent tagging issues with custom tags

### ISSUE THREAD

**Issue by thegoatherder** [spencermountain/compromise#996]
I'm trying to understand what is happening with the tagger in this example. This seems to be a regression in Compromise as it was previously working in v13.

A tag called `#Diagnostic` is introduced via plugin. It sometimes works and sometimes doesn't. In all cases the tag words are being recognised as `#Noun` and `#Singular` but not always as a `#Diagnostic`

```javascript

const nlp = require('compromise')
nlp.verbose('tagger')
// Add the #Diagnostic tag and assign 2 words to it
nlp.plugin({
  tags: {
    Diagnostic: {
      isA: 'Noun',
    },
  },
  words: {
    echo: 'Diagnostic',
    scan: 'Diagnostic',
  },
})

// ❌ `scan` tagged Noun, Singular
nlp(`I think that she should have a further scan in a year's time.`).debug()
// ✅ `scan` tagged Noun, Diagnostic, Singular
nlp('She went for a scan.').debug()
// ❌ `echo` tagged Noun, Singular
nlp('The plan is for a further echo in October this year and then I will write regarding further follow-up.').debug()
// ✅ `echo` tagged Noun, Diagnostic, Singular
nlp('He had an echo today.').debug()

```

## Output:

```ascii
 I           → #Pronoun                 1-lexicon
 think       → #Infinitive              1-lexicon
 that        → #Determiner              1-lexicon
 she         → #Pronoun                 1-lexicon
 should      → #Modal                   1-lexicon
 have        → #Infinitive              1-lexicon
 a           → #Determiner              1-lexicon
 further     → #Adverb                  1-lexicon
 scan        → #Diagnostic              1-lexicon
 in          → #Preposition             1-lexicon
 a           → #Determiner              1-lexicon
 year's      → #Duration                1-lexicon
 time        → #Singular                1-lexicon
 I           → #Noun                     -inferred by #Pronoun
 think       → #Verb                     -inferred by #PresentTense
 think       → #Verb, #PresentTense      -inferred by #Infinitive
 she         → #Noun                     -inferred by #Pronoun
 should      → #Verb                     -inferred by #Modal
 have        → #Verb                     -inferred by #PresentTense
 have        → #Verb, #PresentTense      -inferred by #Infinitive
 scan        → #Noun                     -inferred by #Diagnostic
 scan        → #Singular                3-singular-guess
 year's      → #Date, #Noun              -inferred by #Duration
 time        → #Noun                     -inferred by #Singular
        ↓ - 'further' (#Adverb)  
 scan        → #Infinitive              3-[switch] (Noun|Verb)
 scan        → #Verb                     -inferred by #PresentTense
 scan        → #Verb, #PresentTense      -inferred by #Infinitive
        ↓ - 'year's' (#Date)  
 time        → #Singular                3-[switch] (Noun|Verb)
 time        → #Noun                     -inferred by #Singular
 year's      → #Possessive


  → 4 post-tagger:
 a           → #Value                   [post] 'a-is-one'
 scan        → #Noun                    [post] 'the-only-reason'
 scan        → #Singular                quick-singular
 should      → #Auxiliary               [post] 'modal-verb'
 that        → #Conjunction             [post] 'said-that-he'


  → 1 post-tagger:
 She         → #Pronoun                 1-lexicon
 went        → #PastTense               1-lexicon
 for         → #Preposition             1-lexicon
 a           → #Determiner              1-lexicon
 scan        → #Diagnostic              1-lexicon
 She         → #Noun                     -inferred by #Pronoun
 went        → #Verb                     -inferred by #PastTense
 scan        → #Noun                     -inferred by #Diagnostic
 scan        → #Singular                3-singular-guess
        ↓ - 'a' (#Determiner)  
 scan        → #Singular                3-[switch] (Noun|Verb)
 scan        → #Noun                     -inferred by #Diagnostic
 scan        → #Noun                     -inferred by #Singular
 He          → #Pronoun                 1-lexicon
 had         → #PastTense               1-lexicon
 an          → #Determiner              1-lexicon
 echo        → #Diagnostic              1-lexicon
 today       → #Date                    1-lexicon
 He          → #Noun                     -inferred by #Pronoun
 had         → #Verb                     -inferred by #PastTense
 echo        → #Noun                     -inferred by #Diagnostic
 echo        → #Singular                3-singular-guess
        ↓ - 'an' (#Determiner)  
 echo        → #Singular                3-[switch] (Noun|Verb)
 echo        → #Noun                     -inferred by #Diagnostic
 echo        → #Noun                     -inferred by #Singular
 The         → #Determiner              1-lexicon
 plan        → #Infinitive              1-lexicon
 is          → #Copula, #PresentTense   1-lexicon
 is          → #Copula
 is          → #PresentTense
 for         → #Preposition             1-lexicon
 a           → #Determiner              1-lexicon
 further     → #Adverb                  1-lexicon
 echo        → #Diagnostic              1-lexicon
 in          → #Preposition             1-lexicon
 October     → #Month                   1-lexicon
 this        → #Determiner              1-lexicon
 year        → #Duration                1-lexicon
 and         → #Conjunction             1-lexicon
 then        → #Adverb                  1-lexicon
 I           → #Pronoun                 1-lexicon
 will        → #Modal                   1-lexicon
 write       → #Infinitive              1-lexicon
 regarding   → #Gerund                  1-lexicon
 further     → #Adverb                  1-lexicon
 follow up   → #PhrasalVerb, #Infinitive  1-multi-lexicon
 follow up   → #PhrasalVerb
 follow up   → #Infinitive
 up          → #Particle                1-phrasal-particle
 plan        → #Verb                     -inferred by #PresentTense
 plan        → #Verb, #PresentTense      -inferred by #Infinitive
 is          → #Verb                     -inferred by #Copula
 is          → #Verb                     -inferred by #PresentTense
 echo        → #Noun                     -inferred by #Diagnostic
 echo        → #Singular                3-singular-guess
 October     → #Date, #Noun              -inferred by #Month
 year        → #Date, #Noun              -inferred by #Duration
 I           → #Noun                     -inferred by #Pronoun
 will        → #Verb                     -inferred by #Modal
 write       → #Verb                     -inferred by #PresentTense
 write       → #Verb, #PresentTense      -inferred by #Infinitive
 regarding   → #Verb                     -inferred by #PresentTense
 regarding   → #Verb, #PresentTense      -inferred by #Gerund
 follow      → #Verb                     -inferred by #PhrasalVerb
 follow      → #Verb                     -inferred by #PresentTense
 follow      → #Verb, #PresentTense      -inferred by #Infinitive
 up          → #Verb                     -inferred by #PhrasalVerb
 up          → #Verb, #PhrasalVerb       -inferred by #Particle

       ↓ - 'is' 
 plan        → #Singular                3-[switch] (Noun|Verb)
 plan        → #Noun                     -inferred by #Singular
        ↓ - 'further' (#Adverb)  
 echo        → #Infinitive              3-[switch] (Noun|Verb)
 echo        → #Verb                     -inferred by #PresentTense
 echo        → #Verb, #PresentTense      -inferred by #Infinitive
 follow up   → #Hyphenated              1-punct-hyphen''


  → 5 post-tagger:
 will        → #Modal                   [post] 'will-verb'
 will        → #Auxiliary               [post] 'have-had'
 echo        → #Noun                    [post] 'the-only-reason'
 echo        → #Singular                quick-singular
 will        → #Auxiliary               [post] 'modal-verb'
 follow up   → #PhrasalVerb             [post] 'foo-up'
[nodemon] clean exit - waiting for changes before restart
[nodemon] restarting due to changes...
[nodemon] starting `node ./scratch.js`
 I           → #Pronoun                 1-lexicon
 think       → #Infinitive              1-lexicon
 that        → #Determiner              1-lexicon
 she         → #Pronoun                 1-lexicon
 should      → #Modal                   1-lexicon
 have        → #Infinitive              1-lexicon
 a           → #Determiner              1-lexicon
 further     → #Adverb                  1-lexicon
 scan        → #Diagnostic              1-lexicon
 in          → #Preposition             1-lexicon
 a           → #Determiner              1-lexicon
 year's      → #Duration                1-lexicon
 time        → #Singular                1-lexicon
 I           → #Noun                     -inferred by #Pronoun
 think       → #Verb                     -inferred by #PresentTense
 think       → #Verb, #PresentTense      -inferred by #Infinitive
 she         → #Noun                     -inferred by #Pronoun
 should      → #Verb                     -inferred by #Modal
 have        → #Verb                     -inferred by #PresentTense
 have        → #Verb, #PresentTense      -inferred by #Infinitive
 scan        → #Noun                     -inferred by #Diagnostic
 scan        → #Singular                3-singular-guess
 year's      → #Date, #Noun              -inferred by #Duration
 time        → #Noun                     -inferred by #Singular
        ↓ - 'further' (#Adverb)  
 scan        → #Infinitive              3-[switch] (Noun|Verb)
 scan        → #Verb                     -inferred by #PresentTense
 scan        → #Verb, #PresentTense      -inferred by #Infinitive
        ↓ - 'year's' (#Date)  
 time        → #Singular                3-[switch] (Noun|Verb)
 time        → #Noun                     -inferred by #Singular
 year's      → #Possessive


  → 4 post-tagger:
 a           → #Value                   [post] 'a-is-one'
 scan        → #Noun                    [post] 'the-only-reason'
 scan        → #Singular                quick-singular
 should      → #Auxiliary               [post] 'modal-verb'
 that        → #Conjunction             [post] 'said-that-he'


  → 1 post-tagger:

  ┌─────────
  │ 'I'        - Noun, Pronoun
  │ 'think'    - Verb, PresentTense, Infinitive
  │ 'that'     - Conjunction
  │ 'she'      - Noun, Pronoun
  │ 'should'   - Verb, Modal, Auxiliary
  │ 'have'     - Verb, PresentTense, Infinitive
  │ 'a'        - Determiner
  │ 'further'  - Adverb
  │ 'scan'     - Noun, Singular
  │ 'in'       - Preposition
  │ 'a'        - Determiner, Value
  │ 'year's'   - Date, Noun, Duration, Possessive
  │ 'time'     - Noun, Singular


 She         → #Pronoun                 1-lexicon
 went        → #PastTense               1-lexicon
 for         → #Preposition             1-lexicon
 a           → #Determiner              1-lexicon
 scan        → #Diagnostic              1-lexicon
 She         → #Noun                     -inferred by #Pronoun
 went        → #Verb                     -inferred by #PastTense
 scan        → #Noun                     -inferred by #Diagnostic
 scan        → #Singular                3-singular-guess
        ↓ - 'a' (#Determiner)  
 scan        → #Singular                3-[switch] (Noun|Verb)
 scan        → #Noun                     -inferred by #Diagnostic
 scan        → #Noun                     -inferred by #Singular

  ┌─────────
  │ 'She'      - Noun, Pronoun
  │ 'went'     - Verb, PastTense
  │ 'for'      - Preposition
  │ 'a'        - Determiner
  │ 'scan'     - Noun, Diagnostic, Singular


 He          → #Pronoun                 1-lexicon
 had         → #PastTense               1-lexicon
 an          → #Determiner              1-lexicon
 echo        → #Diagnostic              1-lexicon
 today       → #Date                    1-lexicon
 He          → #Noun                     -inferred by #Pronoun
 had         → #Verb                     -inferred by #PastTense
 echo        → #Noun                     -inferred by #Diagnostic
 echo        → #Singular                3-singular-guess
        ↓ - 'an' (#Determiner)  
 echo        → #Singular                3-[switch] (Noun|Verb)
 echo        → #Noun                     -inferred by #Diagnostic
 echo        → #Noun                     -inferred by #Singular

  ┌─────────
  │ 'He'       - Noun, Pronoun
  │ 'had'      - Verb, PastTense
  │ 'an'       - Determiner
  │ 'echo'     - Noun, Diagnostic, Singular
  │ 'today'    - Date


 The         → #Determiner              1-lexicon
 plan        → #Infinitive              1-lexicon
 is          → #Copula, #PresentTense   1-lexicon
 is          → #Copula
 is          → #PresentTense
 for         → #Preposition             1-lexicon
 a           → #Determiner              1-lexicon
 further     → #Adverb                  1-lexicon
 echo        → #Diagnostic              1-lexicon
 in          → #Preposition             1-lexicon
 October     → #Month                   1-lexicon
 this        → #Determiner              1-lexicon
 year        → #Duration                1-lexicon
 and         → #Conjunction             1-lexicon
 then        → #Adverb                  1-lexicon
 I           → #Pronoun                 1-lexicon
 will        → #Modal                   1-lexicon
 write       → #Infinitive              1-lexicon
 regarding   → #Gerund                  1-lexicon
 further     → #Adverb                  1-lexicon
 follow up   → #PhrasalVerb, #Infinitive  1-multi-lexicon
 follow up   → #PhrasalVerb
 follow up   → #Infinitive
 up          → #Particle                1-phrasal-particle
 plan        → #Verb                     -inferred by #PresentTense
 plan        → #Verb, #PresentTense      -inferred by #Infinitive
 is          → #Verb                     -inferred by #Copula
 is          → #Verb                     -inferred by #PresentTense
 echo        → #Noun                     -inferred by #Diagnostic
 echo        → #Singular                3-singular-guess
 October     → #Date, #Noun              -inferred by #Month
 year        → #Date, #Noun              -inferred by #Duration
 I           → #Noun                     -inferred by #Pronoun
 will        → #Verb                     -inferred by #Modal
 write       → #Verb                     -inferred by #PresentTense
 write       → #Verb, #PresentTense      -inferred by #Infinitive
 regarding   → #Verb                     -inferred by #PresentTense
 regarding   → #Verb, #PresentTense      -inferred by #Gerund
 follow      → #Verb                     -inferred by #PhrasalVerb
 follow      → #Verb                     -inferred by #PresentTense
 follow      → #Verb, #PresentTense      -inferred by #Infinitive
 up          → #Verb                     -inferred by #PhrasalVerb
 up          → #Verb, #PhrasalVerb       -inferred by #Particle

       ↓ - 'is' 
 plan        → #Singular                3-[switch] (Noun|Verb)
 plan        → #Noun                     -inferred by #Singular
        ↓ - 'further' (#Adverb)  
 echo        → #Infinitive              3-[switch] (Noun|Verb)
 echo        → #Verb                     -inferred by #PresentTense
 echo        → #Verb, #PresentTense      -inferred by #Infinitive
 follow up   → #Hyphenated              1-punct-hyphen''


  → 5 post-tagger:
 will        → #Modal                   [post] 'will-verb'
 will        → #Auxiliary               [post] 'have-had'
 echo        → #Noun                    [post] 'the-only-reason'
 echo        → #Singular                quick-singular
 will        → #Auxiliary               [post] 'modal-verb'
 follow up   → #PhrasalVerb             [post] 'foo-up'

  ┌─────────
  │ 'The'      - Determiner
  │ 'plan'     - Noun, Singular
  │ 'is'       - Verb, Copula, PresentTense
  │ 'for'      - Preposition
  │ 'a'        - Determiner
  │ 'further'  - Adverb
  │ 'echo'     - Noun, Singular
  │ 'in'       - Preposition
  │ 'October'  - Date, Noun, Month
  │ 'this'     - Determiner
  │ 'year'     - Date, Noun, Duration
  │ 'and'      - Conjunction
  │ 'then'     - Adverb
  │ 'I'        - Noun, Pronoun
  │ 'will'     - Verb, Modal, Auxiliary
  │ 'write'    - Verb, PresentTense, Infinitive
  │ 'regarding'  - Verb, PresentTense, Gerund
  │ 'further'  - Adverb
  │ 'follow'   - Verb, PhrasalVerb, PresentTense, Infinitive, Hyphenated
  │ 'up'       - Verb, PhrasalVerb, Particle, Hyphenated


[nodemon] clean exit - waiting for changes before restart
[nodemon] restarting due to changes...
[nodemon] starting `node ./scratch.js`
 I           → #Pronoun                 1-lexicon
 think       → #Infinitive              1-lexicon
 that        → #Determiner              1-lexicon
 she         → #Pronoun                 1-lexicon
 should      → #Modal                   1-lexicon
 have        → #Infinitive              1-lexicon
 a           → #Determiner              1-lexicon
 further     → #Adverb                  1-lexicon
 scan        → #Diagnostic              1-lexicon
 in          → #Preposition             1-lexicon
 a           → #Determiner              1-lexicon
 year's      → #Duration                1-lexicon
 time        → #Singular                1-lexicon
 I           → #Noun                     -inferred by #Pronoun
 think       → #Verb                     -inferred by #PresentTense
 think       → #Verb, #PresentTense      -inferred by #Infinitive
 she         → #Noun                     -inferred by #Pronoun
 should      → #Verb                     -inferred by #Modal
 have        → #Verb                     -inferred by #PresentTense
 have        → #Verb, #PresentTense      -inferred by #Infinitive
 scan        → #Noun                     -inferred by #Diagnostic
 scan        → #Singular                3-singular-guess
 year's      → #Date, #Noun              -inferred by #Duration
 time        → #Noun                     -inferred by #Singular
        ↓ - 'further' (#Adverb)  
 scan        → #Infinitive              3-[switch] (Noun|Verb)
 scan        → #Verb                     -inferred by #PresentTense
 scan        → #Verb, #PresentTense      -inferred by #Infinitive
        ↓ - 'year's' (#Date)  
 time        → #Singular                3-[switch] (Noun|Verb)
 time        → #Noun                     -inferred by #Singular
 year's      → #Possessive


  → 4 post-tagger:
 a           → #Value                   [post] 'a-is-one'
 scan        → #Noun                    [post] 'the-only-reason'
 scan        → #Singular                quick-singular
 should      → #Auxiliary               [post] 'modal-verb'
 that        → #Conjunction             [post] 'said-that-he'


  → 1 post-tagger:

  ┌─────────
  │ 'I'        - Noun, Pronoun
  │ 'think'    - Verb, PresentTense, Infinitive
  │ 'that'     - Conjunction
  │ 'she'      - Noun, Pronoun
  │ 'should'   - Verb, Modal, Auxiliary
  │ 'have'     - Verb, PresentTense, Infinitive
  │ 'a'        - Determiner
  │ 'further'  - Adverb
  │ 'scan'     - Noun, Singular
  │ 'in'       - Preposition
  │ 'a'        - Determiner, Value
  │ 'year's'   - Date, Noun, Duration, Possessive
  │ 'time'     - Noun, Singular


 She         → #Pronoun                 1-lexicon
 went        → #PastTense               1-lexicon
 for         → #Preposition             1-lexicon
 a           → #Determiner              1-lexicon
 scan        → #Diagnostic              1-lexicon
 She         → #Noun                     -inferred by #Pronoun
 went        → #Verb                     -inferred by #PastTense
 scan        → #Noun                     -inferred by #Diagnostic
 scan        → #Singular                3-singular-guess
        ↓ - 'a' (#Determiner)  
 scan        → #Singular                3-[switch] (Noun|Verb)
 scan        → #Noun                     -inferred by #Diagnostic
 scan        → #Noun                     -inferred by #Singular

  ┌─────────
  │ 'She'      - Noun, Pronoun
  │ 'went'     - Verb, PastTense
  │ 'for'      - Preposition
  │ 'a'        - Determiner
  │ 'scan'     - Noun, Diagnostic, Singular


 The         → #Determiner              1-lexicon
 plan        → #Infinitive              1-lexicon
 is          → #Copula, #PresentTense   1-lexicon
 is          → #Copula
 is          → #PresentTense
 for         → #Preposition             1-lexicon
 a           → #Determiner              1-lexicon
 further     → #Adverb                  1-lexicon
 echo        → #Diagnostic              1-lexicon
 in          → #Preposition             1-lexicon
 October     → #Month                   1-lexicon
 this        → #Determiner              1-lexicon
 year        → #Duration                1-lexicon
 and         → #Conjunction             1-lexicon
 then        → #Adverb                  1-lexicon
 I           → #Pronoun                 1-lexicon
 will        → #Modal                   1-lexicon
 write       → #Infinitive              1-lexicon
 regarding   → #Gerund                  1-lexicon
 further     → #Adverb                  1-lexicon
 follow up   → #PhrasalVerb, #Infinitive  1-multi-lexicon
 follow up   → #PhrasalVerb
 follow up   → #Infinitive
 up          → #Particle                1-phrasal-particle
 plan        → #Verb                     -inferred by #PresentTense
 plan        → #Verb, #PresentTense      -inferred by #Infinitive
 is          → #Verb                     -inferred by #Copula
 is          → #Verb                     -inferred by #PresentTense
 echo        → #Noun                     -inferred by #Diagnostic
 echo        → #Singular                3-singular-guess
 October     → #Date, #Noun              -inferred by #Month
 year        → #Date, #Noun              -inferred by #Duration
 I           → #Noun                     -inferred by #Pronoun
 will        → #Verb                     -inferred by #Modal
 write       → #Verb                     -inferred by #PresentTense
 write       → #Verb, #PresentTense      -inferred by #Infinitive
 regarding   → #Verb                     -inferred by #PresentTense
 regarding   → #Verb, #PresentTense      -inferred by #Gerund
 follow      → #Verb                     -inferred by #PhrasalVerb
 follow      → #Verb                     -inferred by #PresentTense
 follow      → #Verb, #PresentTense      -inferred by #Infinitive
 up          → #Verb                     -inferred by #PhrasalVerb
 up          → #Verb, #PhrasalVerb       -inferred by #Particle

       ↓ - 'is' 
 plan        → #Singular                3-[switch] (Noun|Verb)
 plan        → #Noun                     -inferred by #Singular
        ↓ - 'further' (#Adverb)  
 echo        → #Infinitive              3-[switch] (Noun|Verb)
 echo        → #Verb                     -inferred by #PresentTense
 echo        → #Verb, #PresentTense      -inferred by #Infinitive
 follow up   → #Hyphenated              1-punct-hyphen''


  → 5 post-tagger:
 will        → #Modal                   [post] 'will-verb'
 will        → #Auxiliary               [post] 'have-had'
 echo        → #Noun                    [post] 'the-only-reason'
 echo        → #Singular                quick-singular
 will        → #Auxiliary               [post] 'modal-verb'
 follow up   → #PhrasalVerb             [post] 'foo-up'

  ┌─────────
  │ 'The'      - Determiner
  │ 'plan'     - Noun, Singular
  │ 'is'       - Verb, Copula, PresentTense
  │ 'for'      - Preposition
  │ 'a'        - Determiner
  │ 'further'  - Adverb
  │ 'echo'     - Noun, Singular
  │ 'in'       - Preposition
  │ 'October'  - Date, Noun, Month
  │ 'this'     - Determiner
  │ 'year'     - Date, Noun, Duration
  │ 'and'      - Conjunction
  │ 'then'     - Adverb
  │ 'I'        - Noun, Pronoun
  │ 'will'     - Verb, Modal, Auxiliary
  │ 'write'    - Verb, PresentTense, Infinitive
  │ 'regarding'  - Verb, PresentTense, Gerund
  │ 'further'  - Adverb
  │ 'follow'   - Verb, PhrasalVerb, PresentTense, Infinitive, Hyphenated
  │ 'up'       - Verb, PhrasalVerb, Particle, Hyphenated


 He          → #Pronoun                 1-lexicon
 had         → #PastTense               1-lexicon
 an          → #Determiner              1-lexicon
 echo        → #Diagnostic              1-lexicon
 today       → #Date                    1-lexicon
 He          → #Noun                     -inferred by #Pronoun
 had         → #Verb                     -inferred by #PastTense
 echo        → #Noun                     -inferred by #Diagnostic
 echo        → #Singular                3-singular-guess
        ↓ - 'an' (#Determiner)  
 echo        → #Singular                3-[switch] (Noun|Verb)
 echo        → #Noun                     -inferred by #Diagnostic
 echo        → #Noun                     -inferred by #Singular

  ┌─────────
  │ 'He'       - Noun, Pronoun
  │ 'had'      - Verb, PastTense
  │ 'an'       - Determiner
  │ 'echo'     - Noun, Diagnostic, Singular
  │ 'today'    - Date
```

**Comment by spencermountain** [comment]
hey - yeah, 'scan' and 'echo' are both noun/verbs and they're getting flipped around by the tagger. When this happens we lose the user-given tag. 

agree it's an awkward outcome, not sure how best to fix it

**Comment by thegoatherder** [comment]
Hmmm. But it’s a user tag and I’ve forced it to Noun. Shouldn’t that get
priority over core tagging? --

-----------------------------------------------------------
"Are you sleep walking through your waking life
      ...or wake walking through your dreams?"

**Comment by spencermountain** [comment]
it's a verb for a while, because the Adverb - like 'quickly scan'
cheers

**Comment by spencermountain** [comment]
adding {further:'Adjective'} will fix this example

**Comment by thegoatherder** [comment]
@spencermountain I have been experimenting with this. I have had some success by defining `#Diagnostic` as both a `#Verb` and a `#Noun`, but I get run-time console warnings from `grad-school` when doing so. I've tried some other methods with less success. 

## Method 1: Define `#Diagnostic` as both `#Noun` and `#Verb`

```javascript
const nlp = require('compromise')
nlp.plugin({
  tags: { Diagnostic: { isA: ['Noun', 'Verb'] } },
  words: {
    scan: 'Diagnostic',
    echo: 'Diagnostic',
  },
})

console.log('✅ works')
nlp('I will scan him tomorrow').debug()
nlp('I gave him a scan').debug()
nlp('Yesterday he had an echo').debug()
nlp('We will perform an echo this evening').debug()

console.log('❌ does not work')
nlp('Plan: Echo').debug()
```
### Output

As you can see there is a warning from `grad-school` and the last debug output is not correctly tagging `Echo`. All other outputs work but the console warning pollutes my Jest tests and runtime output - not sure how to suppress.

```
[Grad] - missing node 'Noun,Verb'

✅ works
  ┌─────────
  │ 'I'        - Noun, Pronoun
  │ 'will'     - Verb, Modal, Auxiliary
  │ 'scan'     - Diagnostic, Verb, PresentTense, Infinitive
  │ 'him'      - Noun, Pronoun
  │ 'tomorrow'  - Date



  ┌─────────
  │ 'I'        - Noun, Pronoun
  │ 'gave'     - Verb, PastTense
  │ 'him'      - Noun, Pronoun
  │ 'a'        - Determiner
  │ 'scan'     - Diagnostic, Noun, Singular



  ┌─────────
  │ 'Yesterday'  - Date
  │ 'he'       - Noun, Pronoun
  │ 'had'      - Verb, PastTense
  │ 'an'       - Determiner
  │ 'echo'     - Diagnostic, Noun, Singular



  ┌─────────
  │ 'We'       - Noun, Pronoun
  │ 'will'     - Verb, Modal, Auxiliary
  │ 'perform'  - Verb, PresentTense, Infinitive
  │ 'an'       - Determiner
  │ 'echo'     - Diagnostic, Noun, Singular
  │ 'this'     - Determiner
  │ 'evening'  - Noun, Singular

❌ does not work

  ┌─────────
  │ 'Plan'     - Verb, PresentTense, Infinitive
  │ 'Echo'     - Verb, PresentTense, Infinitive
```

## Method 2: using the `also` array

Using the `also` definition in the plugin - the warning disappears but there is 100% failure rate at tagging my words

```javascript
const nlp = require('compromise')
nlp.plugin({
  tags: { Diagnostic: { isA: ['Noun'], also: ['Verb'] } },
  words: {
    scan: 'Diagnostic',
    echo: 'Diagnostic',
  },
})

console.log('❌ none of these work')
nlp('I will scan him tomorrow').debug()
nlp('I gave him a scan').debug()
nlp('Yesterday he had an echo').debug()
nlp('We will perform an echo this evening').debug()
nlp('Plan: Echo').debug()
```

### Output

```
❌ none of these work
┌─────────
  │ 'I'        - Noun, Pronoun
  │ 'will'     - Verb, Modal, Auxiliary
  │ 'scan'     - Verb, PresentTense, Infinitive
  │ 'him'      - Noun, Pronoun
  │ 'tomorrow'  - Date



  ┌─────────
  │ 'I'        - Noun, Pronoun
  │ 'gave'     - Verb, PastTense
  │ 'him'      - Noun, Pronoun
  │ 'a'        - Determiner
  │ 'scan'     - Noun, Singular



  ┌─────────
  │ 'Yesterday'  - Date
  │ 'he'       - Noun, Pronoun
  │ 'had'      - Verb, PastTense
  │ 'an'       - Determiner
  │ 'echo'     - Noun, Singular



  ┌─────────
  │ 'We'       - Noun, Pronoun
  │ 'will'     - Verb, Modal, Auxiliary
  │ 'perform'  - Verb, PresentTense, Infinitive
  │ 'an'       - Determiner
  │ 'echo'     - Noun, Singular
  │ 'this'     - Determiner
  │ 'evening'  - Noun, Singular



  ┌─────────
  │ 'Plan'     - Verb, PresentTense, Infinitive
  │ 'Echo'     - Verb, PresentTense, Infinitive
```

### Method 3: Force notA Verb

```javascript
const nlp = require('compromise')
nlp.plugin({
  tags: { Diagnostic: { isA: ['Noun'], notA: ['Verb'] } },
  words: {
    scan: 'Diagnostic',
    echo: 'Diagnostic',
  },
})

console.log('✅ works')
nlp('I gave him a scan').debug()
nlp('Yesterday he had an echo').debug()
nlp('We will perform an echo this evening').debug()

console.log('❌ does not work')
nlp('I will scan him tomorrow').debug()
nlp('Plan: Echo').debug()
```

### Output

```
✅ works

  ┌─────────
  │ 'I'        - Noun, Pronoun
  │ 'gave'     - Verb, PastTense
  │ 'him'      - Noun, Pronoun
  │ 'a'        - Determiner
  │ 'scan'     - Noun, Diagnostic, Singular



  ┌─────────
  │ 'Yesterday'  - Date
  │ 'he'       - Noun, Pronoun
  │ 'had'      - Verb, PastTense
  │ 'an'       - Determiner
  │ 'echo'     - Noun, Diagnostic, Singular



  ┌─────────
  │ 'We'       - Noun, Pronoun
  │ 'will'     - Verb, Modal, Auxiliary
  │ 'perform'  - Verb, PresentTense, Infinitive
  │ 'an'       - Determiner
  │ 'echo'     - Noun, Diagnostic, Singular
  │ 'this'     - Determiner
  │ 'evening'  - Noun, Singular


❌ does not work

  ┌─────────
  │ 'I'        - Noun, Pronoun
  │ 'will'     - Verb, Modal, Auxiliary
  │ 'scan'     - Verb, PresentTense, Infinitive
  │ 'him'      - Noun, Pronoun
  │ 'tomorrow'  - Date



  ┌─────────
  │ 'Plan'     - Verb, PresentTense, Infinitive
  │ 'Echo'     - Verb, PresentTense, Infinitive
```

**Comment by spencermountain** [comment]
this is now fixed in `14.11.0` using the new [freeze](https://observablehq.com/@spencermountain/compromise-freeze) feature
```js
nlp.addWords({    echo: 'Diagnostic',    scan: 'Diagnostic'}, true) // 👈 add true as 2nd param
nlp(`I think that she should have a further scan in a year's time.`).debug() // ✅
nlp('The plan is for a further echo in October').debug() // ✅
```
cheers
